src/bp/ui-studio/src/web/views/OneFlow/diagram/VariableTypesForm/index.tsx

```diff
@@ -45,11 +46,19 @@ const VariableForm: FC<Props> = props => {
   const updateFormItem = (data: sdk.NLU.EntityDefinition) => {
     props.setActiveFormItem({ type: 'variableType', data })
   }
-
-  const allProps = { ...props, updateEntity, deleteEntity, updateFormItem }
-  const Form = props.formData?.type === 'pattern' ? PatternForm : EnumForm
-
-  return <Form {...allProps}></Form>
+  const defaultProps = { updateEntity, deleteEntity, updateFormItem }
+  if (props.formData.type === 'pattern') {
+    const patternProps = { ...props, ...defaultProps }
+    return <PatternForm {...patternProps} />
+  } else {
+    const listEntities = props.entities.filter(e => e.type === 'list')
+    const enumProps = { ...props, ...defaultProps }
+    return <EnumForm allEntities={listEntities} {...enumProps} />
```

Comment:
Why duplicate props? You could just pass "entities" to enumForm, plus "allProps"

Reply:
Right thx